proxy/middleware.go

```diff
@@ -55,16 +55,16 @@ func (p *Proxy) isAuthorized(w http.ResponseWriter, r *http.Request) (bool, erro
 		Method:   "GET",
 		Headers:  map[string]string{},
 		Path:     r.URL.Path,
-		Host:     r.URL.Host,
+		Host:     r.Host,
```

Comment:
Will this have broader implications?

Reply:
No, it makes the forward auth works correctly. Before this change, with request like `http://pomerium/?uri=https://httpbin.localhost.pomerium.io`, we rewrite it into `https://httpbin.localhost.pomerium.io` and send this request as a forward auth request to authorize. In authorize, you can see there's a check that if the request "Host" matches the forward auth "Host".

Also note that this change just affect forward auth mode.